src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -93,22 +139,38 @@ internal static SafeFreeCredentials AcquireCredentialsHandle(string package, boo
 
         internal static void ValidateImpersonationLevel(TokenImpersonationLevel impersonationLevel)
         {
-            throw new PlatformNotSupportedException();
+            if (impersonationLevel != TokenImpersonationLevel.Identification)
+            {
+                throw new ArgumentOutOfRangeException("impersonationLevel", impersonationLevel.ToString(),
+                    SR.net_auth_supported_impl_levels);
+            }
+
         }
 
         internal static void ThrowCredentialException(long error)
         {
-            throw new PlatformNotSupportedException();
+            string message = SR.net_auth_alert;
+            if (error == LogonDenied)
+            {
+                message = SR.net_auth_bad_client_creds;
+            }
+
+            if ((int)error == NegoState.ERROR_TRUST_FAILURE)
+            {
+                message = SR.net_auth_context_expectation_remote;
+            }
+
+            throw new AuthenticationException(message, null);
```

Comment:
@shrutigarg I think this code dates back to when we weren't aware that Win32Exception is supported on Unix. Do we still need this method and IsLogonDeniedException in the PAL implementation?

@bartonjs this is not really a mapping. What I can deduce from the Windows code is that a custom exception message is provided for these 2 errors (presumably they are the most common) and a generic message for all others.